merchants/verkkokauppa/product/types.py

```diff
@@ -37,3 +37,53 @@ def from_json(cls, json: Dict[str, Any]) -> "Product":
             )
         except (KeyError, ValueError) as e:
             raise ParseProductError("Could not parse product") from e
+
+
+@dataclass(frozen=True)
+class CreateOrUpdateAccountingParams:
+    vat_code: str
+    internal_order: str
+    profit_center: str
+    project: str
+    operation_area: str
+    company_code: str
+    main_ledger_account: str
+
+    def to_json(self) -> Dict[str, Any]:
+        return {
+            "vatCode": self.vat_code,
+            "internalOrder": self.internal_order,
+            "profitCenter": self.profit_center,
+            "project": self.project,
+            "operationArea": self.operation_area,
+            "companyCode": self.company_code,
+            "mainLedgerAccount": self.main_ledger_account,
+        }
+
+
+@dataclass(frozen=True)
+class Accounting:
+    product_id: str
+    vat_code: str
+    internal_order: str
+    profit_center: str
+    project: str
+    operation_area: str
+    company_code: str
+    main_ledger_account: str
+
+    @classmethod
+    def from_json(cls, json: Dict[str, Any]) -> "Accounting":
```

Comment:
`-> "Accounting"`? should it be `-> str`?

Reply:
Nope, this takes a JSON (dictionary) and transforms it into `Accounting` object. So the return type is `Accounting`.

I just learned that Python does not allow "typing itself" - meaning using the class name as a type within the class. The workaround seems to be using quotes.

Apparently Python 3.11 has new `Self` type that can be used instead: https://docs.python.org/3/library/typing.html#typing.Self

Reply:
`Self` TIL 🙏